### ISSUE TITLE
Wayland Support

### ISSUE THREAD

**Issue by kaimast** [headsetapp/headset-electron#344]
Hi,

It would be great if this app had Wayland support. Right now the application looks blurry on HighDPI screens on Wayland.

I think Electron 12 does support this feature. 

-Kai

**Comment by fcastilloec** [comment]
Wayland is definitely on the roadmap. Right now, electron 12 is just in beta, and it seems that CSD decorations are having some problems. Gnome requires CSD when using Wayland which is the main GUI we support.
It probably will take until electron 13 (or even 14) to iron out all the bugs before we can actually ship a version with proper Wayland support, otherwise, we might get flooded with issues about it, even though the issues are with electron itself and probably should be reported there. The good news is that development on the electron side is pretty fast, so it might take only a couple of months 😄

**Comment by kaimast** [comment]
Good to know!

Chrome works very well on Wayland even with CSD FWIW. Not sure how much different it is for electron apps.

**Comment by kaimast** [comment]
Electron 12 is out now but, as you said, it probably makes sense to wait until titlebar/headerbar supports lands. Having a headerbar might be a great addition to the app regardless of whether Wayland is used. 

Gonna link these issues here for future reference https://github.com/electron/electron/issues/27016 https://github.com/electron/electron/issues/11907

**Comment by fcastilloec** [comment]
There's a new PR #372 that updates our code to electron v12, since the version we're currently using (v9) has reached end-of-support.
@kaimast once it's merged to master, feel free to test it on Wayland, there are instructions on the README on how to run it (or build it). If you're having trouble, reach me on Slack for help.
You can report back on this issue for any problems you encounter. We're probably won't release a new version yet but any test is greatly appreciated.

**Comment by fcastilloec** [comment]
For anybody looking for a workaround until we officially add support for Wayland. See https://github.com/headsetapp/headset-electron/issues/453#issuecomment-1029685907
